libafl/src/mutators/mod.rs

```diff
@@ -102,18 +103,20 @@ pub trait Mutator<I, S> {
 
 /// A mutator that takes input, and returns a vector of mutated inputs.
 /// Simple as that.
-pub trait MultipleMutator<I, S> {
-    /// Mutate a given input
-    fn mutate(
+pub trait MultiMutator<I, S>: Named {
+    /// Mutate a given input up to `max_count` times,
+    /// or as many times as appropriate, if no `max_count` is given
+    fn multi_mutate(
         &mut self,
         state: &mut S,
         input: &I,
-        vec: &mut Vec<I>,
         stage_idx: i32,
-    ) -> Result<MutationResult, Error>;
+        max_count: Option<usize>,
```

Comment:
@tokatoka should we have a `max_count` here or leave it out? Could make sense if we have a vectorized mutator for example